src/site/page/source.rs

```diff
@@ -12,3 +13,21 @@ pub fn get_filename(file: &str) -> Option<&OsStr> {
     let file_path = Path::new(file);
     file_path.file_stem()
 }
+
+pub fn get_filename_with_dir(file: &str) -> Option<Utf8PathBuf> {
```

Comment:
am slightly worried about error handling/unwraps here though that might be me be overbearing- what do you think?

Reply:
ye, will refactor to return a result